src/components/PokemonInfo.js

```diff
@@ -1,27 +1,82 @@
-import React from "react"
+import React, { useEffect, useState } from "react"
+import PokemonDexInfo from "./PokemonDexInfo"
 
 export default function PokemonInfo({ location }) {
   const { id, name, url } = location.state
-  return (
-    <div className="container">
+  const [pokemonInfo, setPokemonInfo] = useState({})
+  const [isLoaded, setIsLoaded] = useState(false)
+  const [error, setError] = useState(null)
+  
+  useEffect(() => {
+    fetch(url)
+      .then((res) => res.json())
+      .then(
+        (data) => {
+          setPokemonInfo(data)
+          setIsLoaded(true)
+        },
+        (error) => {
+          setIsLoaded(true)
+          setError(error)
+        }
+      )
+  }, [])
+
+  if (error) {
+    return <div>Error: {error.message}</div>
+  } else if (!isLoaded) {
+    return (
+      <div className="text-center">
+        <div className="spinner-border" role="status">
+          <span className="sr-only">Loading...</span>
+        </div>
+      </div>
+    )
+  }  
+   
+  else {
+    return (
+    <div className="container text-center">
       <h1>{name}</h1>
       <p>Numero de dex del pokemon: {id}</p>
+      <p>Experiencia base: {pokemonInfo.base_experience}</p>
+      <p>Altura: {pokemonInfo.height}</p>
       <p>
-        Endpoint original:
-        <a href={url}> {url}</a>
-      </p>
-      <p>
-        La info del dex esta en este endpoint:
-        <a href={`https://pokeapi.co/api/v2/pokemon-species/${id}`}>
-          {" "}
-          {`https://pokeapi.co/api/v2/pokemon-species/${id}`}
-        </a>
-      </p>
-      <p>
-        Hacer que este componente muestre la información del pokemon, su info
-        del pokedex y el o los tipos ademas de su sprite
+      Tipos: {pokemonInfo.types.map((type) => {
```

Comment:
you can use the shorthand syntax if you are not using more variables `pokemonInfo.types.map(type => `${type.type.name} `)`